File: backend/src/models/Guest.js
const mongoose = require("mongoose");

const guestSchema = new mongoose.Schema(
  {
    fullName: { type: String, required: true },
    address: String,
    contactNumber: { type: String, required: true },
    idNumber: { type: String, required: true },
    dateOfBirth: String,
    gender: String,
  },
  { timestamps: true }
);

module.exports = mongoose.model("Guest", guestSchema);

File: java_backend/src/main/java/com/example/demo/model/Guest.java
package com.example.demo.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "guests")
public class Guest {
    @Id
    private String id;
    private String fullName;
    private String address;
    private String contactNumber;
    private String idNumber;
    private String dateOfBirth;
    private String gender;
}
